src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/ASEReadCounter.java

```diff
@@ -253,34 +251,50 @@ protected void calculateLineForSite(final ReadPileup pileup, final String contig
                 continue;
             }
 
-            if (base.getBase() == refAllele)
+            if (base.getBase() == refAllele) {
                 refCount++;
-            else if (base.getBase() == altAllele)
+            }
+            else if (base.getBase() == altAllele) {
                 altCount++;
+            }
             else {
                 otherBasesCount++;
                 continue;
             }
             totalNonFilteredCount++;
         }
 
-        if (totalNonFilteredCount < minDepthOfNonFilteredBases)
-            return;
+        if (totalNonFilteredCount < minDepthOfNonFilteredBases) {
+            return null;
+        }
 
-        String line = contig + separator +
-                position + separator +
-                siteID + separator +
-                refAllele + separator +
-                altAllele + separator +
-                refCount + separator +
-                altCount + separator +
-                totalNonFilteredCount + separator +
-                lowMAPQDepth + separator +
-                lowBaseQDepth + separator +
-                rawDepth + separator +
-                otherBasesCount + separator +
-                improperPairsCount;
-
-        outputStream.println(line);
+        final StringBuilder line = new StringBuilder();
+        line.append(pileup.getLocation().getContig());
```

Comment:
Sorry about this, I know it's ugly, but I think it will save quite a bit of garbage collection churn.

There are a few ways to improve readability and reduce boiler plate here that you might want to think about in the future, 

1.  when it's something like this you can put two statements on one line, ( know, it violates everything you've ever been taught to believer in, I'm sure I've complained about that very thing before... but there are always things so ugly they require exceptions....)

ie.
```
line.append(pileup.getLocation().getContig()); line.append(separator);
line.append(pileup.getLocation().getStart()); line.append(separator);
```

2.  StringBuilders have a "fluent" interface, so you can chain calls to them together.

```
line.append(pileUp.getLocation().getContig()).append(separator)
    .append(pileup.getLocation().getStart()).append(separator)
```